runparams.py

```diff
@@ -3,87 +3,148 @@
-        parser.add_argument("input_dir", help="Directory containing the input specification files")
-        parser.add_argument("output_dir", nargs="?", help="Directory to write the output files to")
+        return {k: getattr(self, k) for k in ("autogen_header",)}
+
+    def check_requirements(self):
+        def check_import_module(module_name, condition):
+            if importlib.util.find_spec(module_name) is None:
+                self.log.error(f"Python module '{module_name}' is required when {condition} is specified.  Make sure it's installed.")
+
+        def check_external_program(program_name, condition):
+            if shutil.which(program_name) is None:
+                self.log.error(
+                    f"Program '{program_name}' is required when {condition} is specified.  Make sure it's installed and present in your PATH."
+                )
+
+        if self.generate_jsondump:
+            check_import_module("jsonpickle", "JSON dump generation")
+        if self.generate_mkdocs:
+            check_import_module("jinja2", "MkDocs generation")
+        if self.generate_rdf:
+            check_import_module("rdflib", "RDF generation")
+        if self.generate_tex:
+            check_external_program("pandoc", "TeX generation")
+            check_import_module("jinja2", "TeX generation")
+
+    def process_args(self, opts=sys.argv[1:]):
+        def check_input_path(p):
+            if not p.exists():
+                raise argparse.ArgumentTypeError(f"Input directory '{p}' does not exist.")
+            if not p.is_dir():
+                raise argparse.ArgumentTypeError(f"Input path '{p}' is not a directory.")
+            if p.name != "model":
+                raise argparse.ArgumentTypeError(f"Input directory '{p}' must be named 'model'.")
+
+        parser = argparse.ArgumentParser(description="Generate documentation from an SPDXv3 model.")
+
+        parser.add_argument("input_dir", type=str, help="Path to the input 'model' directory.")
+
         parser.add_argument("-d", "--debug", action="store_true", help="Print debug output")
-        parser.add_argument("-f", "--force", action="store_true", help="Overwrite existing generated files")
-        parser.add_argument("-n", "--nooutput", action="store_true", help="Do not generate anything, only check input")
-        parser.add_argument("-q", "--quiet", action="store_true", help="Print no output")
+        parser.add_argument("-f", "--force", action="store_true", help="Force overwrite of existing output directories.")
+        parser.add_argument("-j", "--generate-jsondump", action="store_true", help="Generate a dump of the model in JSON format.")
+        parser.add_argument("-J", "--output-jsondump", type=str, help="Output directory for JSON dump file.")
+        parser.add_argument("-m", "--generate-mkdocs", action="store_true", help="Generate mkdocs output.")
+        parser.add_argument("-M", "--output-mkdocs", type=str, help="Output directory for mkdocs files.")
```

Comment:
```suggestion
        parser.add_argument("-m", "--generate-mkdocs", action="store_true", help="Generate MkDocs output.")
        parser.add_argument("-M", "--output-mkdocs", type=str, help="Output directory for MkDocs files.")
```

Keep the stylistic cases. Similar to PlantUML and TeX.